### PR TITLE
feat(suppression): close ADR-044 P1 roadmap (items 1-6)

### PR DESCRIPTION
**Pull request by napetrov**
## Summary

Closes ADR-044's full P1 roadmap (the P0 slice merged in #585): first-class call-graph reachability detection, plus the smaller configuration/plumbing items.

## Motivation

ADR-044's P0 slice shipped reachability-aware suppression but explicitly left six P1 items unscheduled. This PR closes all six.

### P1 items 1-2 — call-graph evidence + new overlay `ChangeKind`

The P0 slice's `MarkReachability` only walked layout/type-graph reachability (`internal_leak.compute_leak_paths`) — it had no access to the L5 semantic call graph, so the ADR's own headline example (a public inline function calling into a removed internal template specialization, with **no** layout evidence at all) was not closed.

- New `internal_leak.compute_call_graph_leak_paths(snap, internal_namespaces)` walks the optional L5 source graph's `DECL_CALLS_DECL`/`DECL_REFERENCES_DECL` edges from every public entry, returning proof paths for internal decls reachable via a pure call/reference edge. Requires an embedded L5 graph (`--sources`/`--build-info`/`--header-graph`); returns `{}` and changes no behavior otherwise.
- `MarkReachability` now consults this as a second, independent evidence source alongside the layout walk.
- New overlay `ChangeKind.INTERNAL_SYMBOL_REQUIRED_BY_PUBLIC_API` (`BREAKING_KINDS`) — the call-graph analogue of `INTERNAL_TYPE_LEAKS_VIA_PUBLIC_API`. Built by new `internal_leak.detect_call_graph_leaks`, wired into the `DetectInternalLeaks` pipeline step. Triggers **only** on a change whose own kind is already `BREAKING_KINDS`/`API_BREAK_KINDS` (artifact-proven) — per the existing authority rule (ADR-028 D3/ADR-041), the graph edge composes with and explains an already-proven break, never manufactures one.

### P1 item 3 — edge semantics (partially closed)

`reachability_kind` gained a third real value, `symbol_availability`, for the call-graph case — no longer a two-value approximation. The finer `DECL_CALLS_DECL` vs. `DECL_REFERENCES_DECL` distinction is preserved as text inside `reachability_proof_path`; a further machine-readable split is left as a candidate future refinement.

### P1 item 4 — structured report fields

`public_reachable`/`reachability_kind`/`reachability_proof_path` now appear as first-class fields in JSON (both `_change_to_dict` and the separate hand-rolled `_to_json_leaf` path) and SARIF `properties` (camelCased), not just prose inside the `suppression_would_hide_public_break` diagnostic.

### P1 item 5 — configurable internal-namespace convention

New `PolicyFile.internal_namespaces: list[str]`, threaded via `PipelineContext.internal_namespaces` through `MarkReachability`/`DetectInternalLeaks`/`DemoteUnreachableInternalChurn`. Deliberately **not** threaded into `DetectNamespacePatterns`'s `experimental_namespaces` — that governs an unrelated convention despite the ADR's roadmap text grouping all four steps together.

### P1 item 6 — remaining `checker.py` suppression call sites

`_filter_suppressed_changes`/`_apply_surface_metrics` now call `SuppressionList.evaluate()` and append the same `suppression_would_hide_public_break` diagnostic the rest of the pipeline already produces.

## Test plan

- [x] New unit tests: `compute_call_graph_leak_paths`/`detect_call_graph_leaks` edge cases (no build source, no relevant edges, call vs. reference edges, non-internal target, no public entry, custom namespace convention, non-breaking-kind exemption, no evidence) — `tests/test_internal_leak.py`
- [x] New tests for `PolicyFile.internal_namespaces` parsing/validation — `tests/test_policy_file.py`
- [x] New pipeline-threading and checker-level suppression-diagnostic tests — `tests/test_reachability_aware_suppression.py`
- [x] New JSON/SARIF structured-field tests — `tests/test_cov95_reporting.py`, `tests/test_sarif.py`
- [x] Full fast suite: `pytest tests/ -m "not integration and not libabigail and not abicc and not slow and not golden"` → 16408 passed, 0 failed
- [x] `ruff check abicheck/ tests/` clean
- [x] `mypy abicheck/` clean (0 errors, matches documented baseline)
- [x] `python scripts/check_ai_readiness.py` → 0 errors (warnings are pre-existing file-size/orphan-doc items unrelated to this change)
- [x] ADR-044 doc updated: status line, new "P1 slice" implementation section, roadmap section marked closed
- [x] Changelog fragments added (`scriv create`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01YbE7rYA3L7Dv1ixr2trxBa

### CODE REVIEW

File: abicheck/post_processing.py [hunk 1]
```diff
@@ -1468,6 +1478,7 @@ def run(
         new: AbiSnapshot,
         suppression: SuppressionList | None = None,
         frozen_namespaces: list[str] | None = None,
+        internal_namespaces: tuple[str, ...] | None = None,
         scope_to_public_surface: bool = False,
```
[the remark this answers is not in the document]
Comment (napetrov) [reply abicheck/post_processing.py:1566]: Good catch, fixed — moved `internal_namespaces` to the end of the parameter list instead of inserting it before `scope_to_public_surface`, so any existing positional caller is unaffected. Will be in the next push.

---
_Generated by [Claude Code](https://claude.ai/code)_
